evm/utils/db.py

```diff
@@ -8,42 +8,50 @@
     EMPTY_SHA3,
 )
 
+from evm.rlp.blocks import BaseBlock
+from evm.rlp.headers import BlockHeader
 
-def make_block_number_to_hash_lookup_key(block_number):
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from evm.db.chain import BaseChainDB  # noqa
+
+
+def make_block_number_to_hash_lookup_key(block_number: int) -> bytes:
     number_to_hash_key = b'block-number-to-hash:%d' % block_number
     return number_to_hash_key
 
 
-def make_block_hash_to_score_lookup_key(block_hash):
+def make_block_hash_to_score_lookup_key(block_hash: str) -> bytes:
     return b'block-hash-to-score:%s' % block_hash
 
 
-def make_transaction_hash_to_data_lookup_key(transaction_hash):
+def make_transaction_hash_to_data_lookup_key(transaction_hash: str) -> bytes:
     '''
     Look up a transaction that is pending, after being issued locally
     '''
     return b'transaction-hash-to-data:%s' % transaction_hash
 
 
-def make_transaction_hash_to_block_lookup_key(transaction_hash):
+def make_transaction_hash_to_block_lookup_key(transaction_hash: str) -> bytes:
     return b'transaction-hash-to-block:%s' % transaction_hash
 
 
-def get_parent_header(block_header, db):
+def get_parent_header(block_header: BlockHeader, db: 'BaseChainDB') -> BaseBlock:
     """
     Returns the header for the parent block.
     """
     return db.get_block_header_by_hash(block_header.parent_hash)
 
 
-def get_block_header_by_hash(block_hash, db):
+def get_block_header_by_hash(block_hash: BlockHeader, db: 'BaseChainDB') -> BaseBlock:
```

Comment:
The two methods above seem to return a block header, yet the return type is `BaseBlock`

Reply:
Yep, screwed that up. Fixed it.